lib/graphql/language/nodes.rb

```diff
@@ -8,6 +8,8 @@ module Nodes
       # - `children` returns all AST nodes attached to this one. Used for tree traversal.
       # - `scalars` returns all scalar (Ruby) values attached to this one. Used for comparing nodes.
       # - `to_query_string` turns an AST node into a GraphQL string
+      DEFAULT_PRINTER = GraphQL::Language::Printer.new
```

Comment:
Can we get away with making a fresh `Printer` every time? I appreciate the thriftiness of reusing one, but I've also seen people do some funny stuff because they don't know that objects are long-lived. If it's not a perf concern, can we do `printer: GraphQL::Language::Printer.new` in the method signature below?